src/hardware/voodoo.cpp

```diff
@@ -4824,15 +4824,11 @@ static void fastfill(voodoo_state *v)
 
 		for (int32_t curscan = v1yclip; curscan < v3yclip; curscan++)
 		{
-			const poly_extent *extent = &extents[curscan - startscanline];
-			int32_t istartx = extent->startx, istopx = extent->stopx;
+			auto extent = &extents[curscan - startscanline];
 
 			/* force start < stop */
-			if (istartx > istopx)
-			{
-				int32_t temp = istartx;
-				istartx = istopx;
-				istopx = temp;
+			if (extent->startx > extent->stopx) {
+				std::swap(extent->startx, extent->stopx);
```

Comment:
I'm a bit worried about fixing i ntended stuff like this that effectively did nothing. Maybe doing nothing is the correct behaviour here and the code is misleading (the person just forgot to remove the effectively dead code that's no longer needed).

Have ypu compared the output after "fixing" this? Is there any difference?

Reply:
There are a couple areas that process face-forward triangles - a slower path and a fastfill path. 

Both have the same logic, and they also have the same comments. Here's the other area (existing code, not from this PR, see line 4234):

```
/* force start < stop */
if (extent.startx >= extent.stopx) {
	if (extent.startx == extent.stopx) continue;
	std::swap(extent.startx, extent.stopx);
}
```

If the triangles are flipped, then the face is transparent, which is the same is not even processing the triangle:

![2023-07-31_15-29](https://github.com/dosbox-staging/dosbox-staging/assets/1557255/70669665-b7ac-432d-988f-243d17951d01)

Unfortunately during testing I wasn't able to find anything that uses the fastfill code path.

But there is evidence in `raster_fastfill` that it expects face-forward triangles:

1. There is no initial sanity check or bail-out "if start > stop"
2. Loops go from `start` to `stop`
3. a pixel-counter would run in reverse if start > stop: `stats.pixels_out += stopx - startx`
4. the most telling: the function compute a new stop value (line 3633) and at that point, if `stop < start` then it immediately bails out.

So in absense of being able to test it, there are three points of evidence that point to the code actually expecting the triangle extent to be "start <= stop".
 - the comment
 - the code itself (try) to do the swap
 - the raster_fastfill function expecting start <= stop plus actual bail out when stop is recomputed.

Reply:
Thanks, given that you've extensively researched the situation, that make this definitely a safe change. I was just lacking all that context only by looking at this commit 😄